frontend/src/components/NewRunParametersV2.tsx

```diff
@@ -137,7 +149,11 @@ function NewRunParametersV2(props: NewRunParametersProps) {
                   nextUpdatedParameters[k] = { string_value: value };
                   setUpdatedParameters(nextUpdatedParameters);
                   if (props.handleParameterChange) {
-                    props.handleParameterChange(nextUpdatedParameters);
+                    let parametersInRealType = {};
+                    Object.entries(nextUpdatedParameters).map(([k, v]) => {
+                      parametersInRealType[k] = inputConverter(v['string_value']);
+                    });
+                    props.handleParameterChange(parametersInRealType);
```

Comment:
How do we validate the type of parameter matches the value? Feel free to add a TODO here for next PR to implement type check.